core/slice.go

```diff
@@ -364,89 +316,94 @@ func (sl *Slice) GetPendingHeader() (*types.Header, error) {
 }
 
 // SubRelayPendingHeader takes a pending header from the sender (ie dominant), updates the phCache with a composited header and relays result to subordinates
-func (sl *Slice) SubRelayPendingHeader(pendingHeader types.PendingHeader, reorg bool) error {
-	nodeCtx := common.NodeLocation.Context()
-
+func (sl *Slice) SubRelayPendingHeader(pendingHeader types.PendingHeader, reorg bool) {
 	sl.phCachemu.Lock()
 	defer sl.phCachemu.Unlock()
+	nodeCtx := common.NodeLocation.Context()
 
 	if nodeCtx == common.REGION_CTX {
 		sl.updatePhCacheFromDom(pendingHeader, common.NodeLocation.Region(), []int{common.PRIME_CTX}, reorg)
 		for i := range sl.subClients {
-			err := sl.subClients[i].SubRelayPendingHeader(context.Background(), sl.phCache[pendingHeader.Termini[common.NodeLocation.Region()]], reorg)
-			if err != nil {
-				log.Warn("SubRelayPendingHeader", "err:", err)
-			}
+			sl.subClients[i].SubRelayPendingHeader(context.Background(), sl.phCache[pendingHeader.Termini[common.NodeLocation.Region()]], reorg)
 		}
 	} else {
 		sl.updatePhCacheFromDom(pendingHeader, common.NodeLocation.Zone(), []int{common.PRIME_CTX, common.REGION_CTX}, reorg)
-		sl.phCache[pendingHeader.Termini[common.NodeLocation.Zone()]].Header.SetLocation(common.NodeLocation)
 		bestPh, exists := sl.phCache[sl.pendingHeader]
 		if exists {
 			sl.miner.worker.pendingHeaderFeed.Send(bestPh.Header)
 		}
 	}
-	return nil
 }
 
-// updatePhCache takes in an externPendingHeader and updates the pending header on the same terminus if the number is greater
-func (sl *Slice) updatePhCache(externPendingHeader types.PendingHeader) {
+// computePhCache takes in an externPendingHeader and updates the pending header on the same terminus if the number is greater
+func (sl *Slice) computePhCache(localPendingHeaderWithTermini types.PendingHeader, domPendingHeader *types.Header, domOrigin bool) types.PendingHeader {
 	nodeCtx := common.NodeLocation.Context()
 
-	var localPendingHeader types.PendingHeader
-	hash := externPendingHeader.Termini[terminiIndex]
-	localPendingHeader, exists := sl.phCache[hash]
+	var cachedPendingHeaderWithTermini types.PendingHeader
+	hash := localPendingHeaderWithTermini.Termini[terminiIndex]
+	cachedPendingHeaderWithTermini, exists := sl.phCache[hash]
+	var newPh *types.Header
```

Comment:
`newPh` is a pointer to a header in our cache, and this pointer gets returned to the caller where it may be modified. Probably want to copy headers?

Reply:
Updated the combinePendingHeader function to return a copy. This is a new commit. Please check.

Reply:
[cd623ef](https://github.com/dominant-strategies/go-quai/pull/494/commits/cd623efaaf75b47e9ad101aa7b81a40dcc649e2a)